code/datums/components/butchering.dm

```diff
@@ -160,4 +170,60 @@
 	if(eater.safety_mode || (eater.machine_stat & (BROKEN|NOPOWER))) //I'm so sorry.
 		return
 	if(victim.stat == DEAD && (victim.butcher_results || victim.guaranteed_butcher_results))
-		Butcher(parent, victim)
+		on_butchering(parent, victim)
+
+/datum/component/butchering/wearable
+
+/datum/component/butchering/wearable/RegisterWithParent()
+	. = ..()
+	RegisterSignal(parent, COMSIG_MECHA_EQUIPMENT_ATTACHED, .proc/enable_butchering)
+	RegisterSignal(parent, COMSIG_MECHA_EQUIPMENT_DETACHED, .proc/disable_butchering)
+	RegisterSignal(parent, COMSIG_ITEM_EQUIPPED, .proc/worn_enable_butchering)
+	RegisterSignal(parent, COMSIG_ITEM_DROPPED, .proc/worn_disable_butchering)
+	RegisterSignal(parent, COMSIG_MECHA_DRILL_MOB, .proc/on_drill)
+
+/datum/component/butchering/wearable/UnregisterFromParent()
+	. = ..()
+	UnregisterSignal(parent, list(
+		COMSIG_MECHA_DRILL_MOB,
+		COMSIG_ITEM_EQUIPPED,
+		COMSIG_ITEM_DROPPED,
+		COMSIG_MECHA_EQUIPMENT_ATTACHED,
+		COMSIG_MECHA_EQUIPMENT_DETACHED,
+	))
+
+///Enables the butchering mechanic for the mob who has equipped us.
+/datum/component/butchering/wearable/proc/enable_butchering(datum/source)
+	SIGNAL_HANDLER
+	butchering_enabled = TRUE
+
+///Disables the butchering mechanic for the mob who has dropped us.
+/datum/component/butchering/wearable/proc/disable_butchering(datum/source)
+	SIGNAL_HANDLER
+	butchering_enabled = FALSE
+
+///Same as enable_butchering but for worn items
+/datum/component/butchering/wearable/proc/worn_enable_butchering(obj/item/source, mob/user, slot)
+	SIGNAL_HANDLER
+	//check if the item is being not worn
+	if(!(slot & source.slot_flags))
+		return
+	butchering_enabled = TRUE
+	RegisterSignal(user, COMSIG_HUMAN_EARLY_UNARMED_ATTACK, .proc/butcher_target)
+
+///Same as disable_butchering but for worn items
+/datum/component/butchering/wearable/proc/worn_disable_butchering(obj/item/source, mob/user)
+	SIGNAL_HANDLER
+	butchering_enabled = FALSE
+	UnregisterSignal(user, COMSIG_HUMAN_EARLY_UNARMED_ATTACK)
+
+///When we are ready to drill through a mob
+/datum/component/butchering/wearable/proc/on_drill(datum/source, obj/vehicle/sealed/mecha/chassis, mob/living/meat)
+	SIGNAL_HANDLER
+	INVOKE_ASYNC(src, .proc/on_butchering, chassis, meat)
+
+/datum/component/butchering/wearable/proc/butcher_target(mob/user, atom/target, proximity)
```

Comment:
```suggestion
///Called through [COMSIG_HUMAN_EARLY_UNARMED_ATTACK], user being the person wearing the component's parent.
/datum/component/butchering/wearable/proc/butcher_target(mob/user, atom/target, proximity)
```

or whatever other ideas you have, I just would like these all to have comments so they aren't confusing to sift through to figure out which one is for what.

This could be less confusing if you split them into ``/wearable`` and ``/mech`` since they don't seem to share much in common, but I'll leave that up to you.

Reply:
honestly yeah

Reply:
i will split it and put enable_butchering and disable_butchering proc on datum/component/butchering

Reply:
done